src/components/Attachments/AttachmentView/AttachmentViewPdf/BaseAttachmentViewPdf.js

```diff
@@ -1,48 +1,70 @@
+import PropTypes from 'prop-types';
 import React, {memo, useCallback, useContext, useEffect} from 'react';
 import AttachmentCarouselPagerContext from '@components/Attachments/AttachmentCarousel/Pager/AttachmentCarouselPagerContext';
 import PDFView from '@components/PDFView';
 import {attachmentViewPdfDefaultProps, attachmentViewPdfPropTypes} from './propTypes';
 
+const baseAttachmentViewPdfPropTypes = {
+    ...attachmentViewPdfPropTypes,
+
+    /** Triggered when the PDF's onScaleChanged event is triggered */
+    onScaleChanged: PropTypes.func,
+};
+
+const baseAttachmentViewPdfDefaultProps = {
+    ...attachmentViewPdfDefaultProps,
+
+    onScaleChanged: undefined,
+};
+
 function BaseAttachmentViewPdf({
     file,
     encryptedSourceUrl,
     isFocused,
     isUsedInCarousel,
-    onPress,
+    onPress: onPressProp,
     onScaleChanged: onScaleChangedProp,
     onToggleKeyboard,
     onLoadComplete,
     errorLabelStyles,
     style,
 }) {
     const attachmentCarouselPagerContext = useContext(AttachmentCarouselPagerContext);
+    const isScrollEnabled = attachmentCarouselPagerContext === null ? undefined : attachmentCarouselPagerContext.isScrollEnabled;
 
     useEffect(() => {
         if (!attachmentCarouselPagerContext) {
             return;
         }
-        attachmentCarouselPagerContext.onPinchGestureChange(false);
+        attachmentCarouselPagerContext.onScaleChanged(1);
         // eslint-disable-next-line react-hooks/exhaustive-deps -- we just want to call this function when component is mounted
     }, []);
 
     const onScaleChanged = useCallback(
-        (scale) => {
-            onScaleChangedProp(scale);
+        (newScale) => {
+            if (onScaleChangedProp !== undefined) {
+                onScaleChangedProp(newScale);
+            }
 
             // When a pdf is shown in a carousel, we want to disable the pager scroll when the pdf is zoomed in
             if (isUsedInCarousel && attachmentCarouselPagerContext) {
-                const shouldPagerScroll = scale === 1;
-
-                attachmentCarouselPagerContext.onPinchGestureChange(!shouldPagerScroll);
+                attachmentCarouselPagerContext.onScaleChanged(newScale);
+            }
+        },
+        [attachmentCarouselPagerContext, isUsedInCarousel, onScaleChangedProp],
+    );
 
-                if (attachmentCarouselPagerContext.shouldPagerScroll.value === shouldPagerScroll) {
-                    return;
-                }
+    const onPress = useCallback(
```

Comment:
Can we also add JS docs here? It's pretty vague what the outcome of this will be since this is onPress and then it will do onTap so it is hard to know what we are trying to accomplish here.